repl/src/main/scala/ammonite/repl/interp/Evaluator.scala

```diff
@@ -206,6 +222,74 @@ object Evaluator{
       )
     }
 
+    def compileCacheBlock(wrapperName: String, code: String, scriptImports: Seq[ImportData]): Res[(Class[_], Seq[ImportData])] = for {
+      _ <- Catching{ case e: ThreadDeath => interrupted() }
+      (classFiles, importData) <- compileClass(
+        s"""
```

Comment:
The structure of the triple-quoted `compileClass` string is also 99% the same, and we probably want to maintain it the same. e.g. if we ever move to class-based-wrappers, we should only need to change it in one place. Currently we'd have to remember to update both string literals to keep things in sync